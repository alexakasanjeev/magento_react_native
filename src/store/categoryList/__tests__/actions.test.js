import {
  setNewCategory,
  getCategoryProducts,
  getCategoryConfigurableProductOptions,
} from '../actions';

describe('categoryList actions', () => {
  test('setNewCategory()', () => {
    // Setup
    const categoryId = '23';

    // Exercise
    const result = setNewCategory(categoryId);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getCategoryProducts()', () => {
    // Setup
    const categoryId = '23';
    const offset = 20;
    const sortOrder = '';

    // Exercise
    const result = getCategoryProducts(categoryId, offset, sortOrder);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getCategoryConfigurableProductOptions()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getCategoryConfigurableProductOptions(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });
});
