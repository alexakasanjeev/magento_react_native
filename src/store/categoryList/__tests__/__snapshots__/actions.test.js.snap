// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categoryList actions getCategoryConfigurableProductOptions() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_CATEGORY_UPDATE_CONF_PRODUCT_REQUEST",
}
`;

exports[`categoryList actions getCategoryProducts() 1`] = `
Object {
  "payload": Object {
    "categoryId": "23",
    "offset": 20,
    "sortOrder": "",
  },
  "type": "MAGENTO_CATEGORY_PRODUCTS_REQUEST",
}
`;

exports[`categoryList actions setNewCategory() 1`] = `
Object {
  "payload": Object {
    "categoryId": "23",
  },
  "type": "SET_NEW_CATEGORY",
}
`;
